model/src/main/kotlin/Identifier.kt

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2017-2018 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.SerializerProvider
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer
+import com.fasterxml.jackson.databind.ser.std.StdSerializer
+
+/**
+ * A unique identifier for a software package.
+ */
+data class Identifier(
+        /**
+         * The name of the package manager that was used to discover this package, for example Maven or NPM.
+         */
+        @JsonProperty("package_manager")
+        val packageManager: String,
+
+        /**
+         * The namespace of the package, for example the group id in Maven or the scope in NPM.
+         */
+        val namespace: String,
+
+        /**
+         * The name of the package.
+         */
+        val name: String,
+
+        /**
+         * The version of the package.
+         */
+        val version: String
+) {
+    companion object {
+        /**
+         * A constant for a [Identifier] where all properties are empty strings.
+         */
+        @JvmField
+        val EMPTY = Identifier(
+                packageManager = "",
+                namespace = "",
+                name = "",
+                version = ""
+        )
+
+        fun fromString(identifier: String): Identifier {
```

Comment:
Please also add docs here to have the class fully documented.

Reply:
Done. I have also changed `matches()` slightly to ignore case when comparing package managers, as we generate those names. For namespace, name, and version I think it is better to not ignore case.